src/main/daml/Daml/Finance/Util/Date/DayCount.daml

```diff
@@ -204,15 +217,32 @@ calcDcf30E360 toDateIsMaturityDate fromDate toDate =
     d1Adj = if d1 == monthDayCount y1 m1 then 30 else d1
     d2Adj = if d2 == 31 || not toDateIsMaturityDate && isEndOfFebruary (y2, m2, d2) then 30 else d2
   in
-    calc30360Formula (y1, m1, d1Adj) (y2, m2, d2Adj)
+    calc30FixFormula (y1, m1, d1Adj) (y2, m2, d2Adj) 360
+  where
+    (y1, m1, d1) = toGregorian fromDate
+    (y2, m2, d2) = toGregorian toDate
+
+-- | Calculate 30E3/360 day count fraction.
+calcDcf30E3360 : Bool -> Date -> Date -> Decimal
+calcDcf30E3360 _ fromDate toDate | fromDate > toDate =
+  error "fromDate is greater than toDate"
+calcDcf30E3360 regularPeriod fromDate toDate =
+  let
+    d1Adj = if d1 == monthDayCount y1 m1 then 30 else d1
+    d2Adj
+      | isEndOfFebruary (y2, m2, d2) && d1Adj > d2 && regularPeriod = d1Adj
+      | otherwise = if d2 == 31 then 30 else d2
+  in
+    calc30FixFormula (y1, m1, d1Adj) (y2, m2, d2Adj) 360
   where
     (y1, m1, d1) = toGregorian fromDate
     (y2, m2, d2) = toGregorian toDate
 
 -- | HIDE
-calc30360Formula : (Int, Month, Int) -> (Int, Month, Int) -> Decimal
-calc30360Formula (y1, m1, d1) (y2, m2, d2) =
-  (/ 360.0) . intToDecimal $ 360 * (y2-y1) + 30 * (fromMonth m2 - fromMonth m1) + (d2-d1)
+calc30FixFormula : (Int, Month, Int) -> (Int, Month, Int) -> Int -> Decimal
+calc30FixFormula (y1, m1, d1) (y2, m2, d2) daysPerYear =
```

Comment:
Same here, it feels more natural to make `daysPerYear` the first input parameter